internal/pkg/cli/storage_init.go

```diff
@@ -0,0 +1,235 @@
+// Copyright Amazon.com, Inc or its affiliates. All rights reserved.
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/config"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+const (
+	dynamoDBStorageType = "Dynamo DB"
+	s3StorageType       = "S3 Bucket"
+)
+
+var storageTypes = []string{
+	dynamoDBStorageType,
+	s3StorageType,
+}
+
+var (
+	fmtStorageInitTypePrompt = "What " + color.Emphasize("type") + " of storage would you like to associate with %s?"
+	storageInitTypeHelp      = "The type of storage you'd like to add to your project. You can choose between key-value (DynamoDB) content (S3) storage types."
+	fmtStorageInitNamePrompt = "What would you like to " + color.Emphasize("name") + " this %s?"
+	storageInitNameHelp      = "The name of this storage resource. You can use the following characters: a-zA-Z0-9-_"
+	storageInitSvcPrompt     = "Which " + color.Emphasize("service") + " would you like to associate with this storage resource?"
+	storageInitSvcHelp       = "The service you'd like to have access to this storage resource. We'll deploy the Cloudformation for the storage when you run `app deploy`."
+)
+
+var fmtStorageInitCreateConfirm = "Okay, we'll create %s %s named %s linked to your %s service."
+
+type s3Config struct {
+	// TODO add website config
+}
+
+type initStorageVars struct {
+	*GlobalOpts
+	StorageType string
+	StorageName string
+	StorageSvc  string
+}
+
+type initStorageOpts struct {
+	initStorageVars
+
+	fs        afero.Fs
+	ws        wsSvcReader
+	store     store
+	appGetter store
+
+	proj *config.Application
+}
+
+func newStorageInitOpts(vars initStorageVars) (*initStorageOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("couldn't connect to application datastore: %w", err)
```

Comment:
I've started writing it like this: `fmt.Errorf("new config store client: %w", err)` or `fmt.Errorf("new workspace client: %w", err)` what do you think?